archive/imo/imo2013_q5.lean

```diff
@@ -0,0 +1,413 @@
+/-
+Copyright (c) 2020 David Renshaw. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Renshaw
+-/
+
+import algebra.geom_sum
+import data.rat.basic
+import data.rat.order
+import data.real.basic
+
+/-!
+# IMO 2013 Q5
+
+Let ℚ>₀ be the set of positive rational numbers. Let f: ℚ>₀ → ℝ be a function satisfying
+the conditions
+
+  (1) f(x) * f(y) ≥ f(x * y)
+  (2) f(x + y)    ≥ f(x) + f(y)
+
+for all x,y ∈ ℚ>₀. Given that f(a) = a for some rational a > 1, prove that f(x) = x for
+all x ∈ ℚ>₀.
+
+# Solution
+
+We provide a direct translation of the solution found in
+https://www.imo-official.org/problems/IMO2013SL.pdf
+-/
+
+open_locale big_operators
+
+lemma factor_xn_minus_yn (x : ℝ) (y : ℝ) (n : ℕ)
+      :  x^n - y^n = (x - y) * (∑ (i:ℕ) in finset.range n, (x ^(i) * y ^(n - 1 - i))) :=
+begin
+  have := geom_sum₂_mul_add (x-y) y n,
+  rw [sub_add_cancel x y, geom_series₂_def] at this,
+  nlinarith,
+end
+
+lemma le_of_all_pow_lt_succ (x y : ℝ) (hx : 1 < x) (hy : 1 < y)
+      (h : ∀n:ℕ, 0 < n → x^n - 1 < y^n)
+      : x ≤ y :=
+begin
+   by_contra hxy,
+   push_neg at hxy,
+   have hxmy : 0 < x - y := sub_pos.mpr hxy,
+   have hn: (∀n:ℕ, 0 < n → (x - y) * (n:ℝ) ≤ x^n - y^n),
+   {
+     intros n hn,
```

Comment:
Braces should not be on their own lines, see the [tactic mode section](https://leanprover-community.github.io/contribute/style.html#tactic-mode) of the style guide.
```suggestion
   { intros n hn,
```
Also, the indentation should be by 2 spaces here, not 3.